app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java

```diff
@@ -449,12 +477,46 @@ public void onActivityResult(int requestCode, int resultCode, Intent intent)
             format = result.getFormatName();
         }
 
-        if(requestCode == SELECT_BARCODE_REQUEST && resultCode == Activity.RESULT_OK)
+        if (resultCode == Activity.RESULT_OK)
         {
-            Log.i(TAG, "Received barcode information from typing it");
+            if(requestCode == SELECT_BARCODE_REQUEST)
+            {
+                Log.i(TAG, "Received barcode information from typing it");
 
-            contents = intent.getStringExtra(BarcodeSelectorActivity.BARCODE_CONTENTS);
-            format = intent.getStringExtra(BarcodeSelectorActivity.BARCODE_FORMAT);
+                contents = intent.getStringExtra(BarcodeSelectorActivity.BARCODE_CONTENTS);
+                format = intent.getStringExtra(BarcodeSelectorActivity.BARCODE_FORMAT);
+            }
+            else if (requestCode == SELECT_IMAGE_REQUEST)
+            {
+                Log.i(TAG, "Received barcode image");
+
+                Bitmap bitmap = null;
+                try {
+                    bitmap = MediaStore.Images.Media.getBitmap(this.getContentResolver(), intent.getData());
+                } catch (IOException e) {
+                    Log.e(TAG, "Error getting the image data");
+                    e.printStackTrace();
+                    return;
+                }
+
+                // In order to decode it, the Bitmap must first be converted into a pixel array...
+                int[] intArray = new int[bitmap.getWidth()*bitmap.getHeight()];
+                bitmap.getPixels(intArray, 0, bitmap.getWidth(), 0, 0, bitmap.getWidth(), bitmap.getHeight());
+
+                // ...and then turned into a binary bitmap from its luminance
+                LuminanceSource source = new RGBLuminanceSource(bitmap.getWidth(), bitmap.getHeight(), intArray);
+                BinaryBitmap binaryBitmap = new BinaryBitmap(new HybridBinarizer(source));
+
+                try {
+                    Result qrCodeResult = new MultiFormatReader().decode(binaryBitmap);
+
+                    contents = qrCodeResult.getText();
+                    format = qrCodeResult.getBarcodeFormat().name();
+                } catch (NotFoundException e) {
+                    Log.i(TAG, "No barcode was found");
```

Comment:
Should this exception be logged? The e.printStackTrace() would not be captured, right?

Reply:
No, I don't believe it should be. All it says is that there was no barcode in the image - which isn't truly an exception.  
Now that I think about it - maybe a toast would be useful?

The printStacktrace() was left there accidentally. I'll remove it.